svm-transaction/src/svm_transaction.rs

```diff
@@ -0,0 +1,26 @@
+use {
+    crate::svm_message::SVMMessage,
+    solana_sdk::{hash::Hash, signature::Signature, transaction::VersionedTransaction},
+};
+
+mod sanitized_transaction;
+
+pub trait SVMTransaction: SVMMessage {
```

Comment:
Just pointing out there's nothing really in the trait (like default implementations) that says this can only be applied to a type that implements `SVMMessage`. Seems like you may have other reasons for this restriction, though?

Reply:
I suppose that's true, but the separation is mainly for convenience.

`SVMTransaction` is intended to be just a `SVMMessage` that is signed. So more that I just do not anticipate it.

Additionally, rust doesn't allow for trait aliases ([yet](https://doc.rust-lang.org/nightly/unstable-book/language-features/trait-alias.html)?). So if we made it so the `SVMTransaction` trait didn't require `SVMMessage` we'd have to specify `impl SVMTransaction + SVMMessage` instead of just `impl SVMTransaction` in all the fns that become generic that currently take `SanitizedTransaction` 

Most of our lower-level functions take `SanitizedMessage` which do not have signatures; if we had a single trait that required us to have some signatures then we'd need some sort of dummy implementation or way around it - or modify all those tests to take a `SanitizedTransaction`.

Reply:
I'd say it's fine as you have it, as long as you don't anticipate anyone ever needing the methods in `SVMTransaction` and _not_ those in `SVMMessage`.